lambdas/es/indexer/index.py

```diff
@@ -187,18 +187,42 @@ def now_like_boto3():
     return datetime.datetime.now(tz=tzutc())
 
 
-def infer_extensions(key, ext):
+def get_compression(ext: str):
+    """return the compression type or None if not supported"""
+    return "gz" if ext == ".gz" else None
+
+
+def get_normalized_extensions(key) -> List[str]:
+    """standard function turning keys into a list of (possibly empty) extensions"""
+    path = pathlib.PurePosixPath(key)
+    try:
+        ext_last = path.suffix.lower()
+        ext_next_last = path.with_suffix('').suffix.lower()
+    except ValueError:
+        return ["", ""]
```

Comment:
Don't think this can actually happen?

Reply:
It happens in only one case that I could find `key="."`, which is evil and why would anyone do that, but S3 will let you have an object named `.` no?